dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -980,64 +982,87 @@ def _DownloadResults(self,
       results.append(data_frame)
     return results
 
+  def _ProcessQuery(
+      self,
+      hostname: str,
+      client: Client,
+      osquery_container: containers.OsqueryQuery
+  ) -> None:
+    """Processes an osquery flow on a GRR client.
+
+    Args:
+      hostname (str): the GRR client hostname.
+      client (Client): the Grr Client.
+      osquery_container (containers.OsqueryQuery): the osquery.
+    """
+    hunt_args = osquery_flows.OsqueryFlowArgs(
+        query=osquery_container.query,
+        timeout_millis=self.timeout_millis,
+        ignore_stderr_errors=self.ignore_stderr_errors)
+
+    try:
+      flow_id = self._LaunchFlow(client, 'OsqueryFlow', hunt_args)
+      self._AwaitFlow(client, flow_id)
+    except DFTimewolfError as error:
+      self.ModuleError(
+          f'Error raised while launching/awaiting flow: {error.message}')
+      return
+
+    name = osquery_container.name
+    description = osquery_container.description
+    query = osquery_container.query
+    flow_identifier = flow_id
+    client_identifier = client.client_id
+
+    results = self._DownloadResults(client, flow_id)
+    if not results:
+      results_container = containers.OsqueryResult(
+          name=name,
+          description=description,
+          query=query,
+          hostname=hostname,
+          data_frame=pd.DataFrame(),
+          flow_identifier=flow_identifier,
+          client_identifier=client_identifier)
+      self.state.StoreContainer(results_container)
```

Comment:
What's the point in storing an empty dataframe here? Wouldn't it be better just to not store any container?

Reply:
I have it as an empty dataframe as the corresponding container attribute is currently not optional.  It also simplifies the logic in downstream processing of the container.

Reply:
My question was more "why add a container at all" if the dataframe is going to be empty anyways.

Reply:
Oops my bad, missed the second question.  My rationale doing it this way that was no result (i.e. empty data) is still a result and is useful feedback downstream to let the module/user know that the query was successful and there was no result.

Reply:
OK, that makes sense, thanks!